**Aim 2**:
We will develop a fragment-level simulator for scRNA-seq data --- while "reasonable" simulators exist for bulk RNA-seq data, very little exists in terms of simulating scRNA-seq data.  Among the simulators that do exist, the focus is mostly on simulating transcript or gene-level _counts_ [(Zappia et al.)](#zappia), rather than the reads themselves. Currently the scRNA-seq technology relies on a range of protocols that differ from each other in various steps, such as selection of fragments (full length, 3' end, 5' end), library preparation protocol (CEL-seq, SMARTer etc.) , cell capture platform (microwell, microfluidic, droplet). All these different approaches introduce different level of noise in the raw data. To effectively simulate these noise parameter our simulator will operate at the level of cDNA fragment selection from raw reference sequence, equivalent to similar tools [(Frazee et al.)](#frazee), popular in bulk RNA-seq data. The simulator will be able to learn parameters from real data, and along with other user-defined parameters, tune the simulation process in a controlled fashion. The sub-aims explains the outline of the simulator, addressing the issues discussed:

- A.) For tag-based protocols, one pre-requisite of analysis of raw scRNA-seq dataset is counting unique molecular identifier-tags (UMI) [(Smith et al.)](#smith). Although UMI-tags plays a crucial role in indentifying the abundance of present isoforms, it has many shortcomings [(Tung et al.)](#tung). Due to shorter length (5-10 bp) the number of different UMI tages (4<sup>5</sup> - 4<sup>10</sup>) is less than the number of molecules present in the cell. This can lead to assigining same UMI-tags to different molecules. On the other hand due to SNP errors, different UMI-s does not always indicate different transcripts, further the presence of multi-mapping and non-uniform frequency of UMI-tags lead to wronng estimation of the transcript count. We will model the problem of UMI-tag assignment as a bipartite matching (cite Figure ?), where all the aforementioned noise can be incorporated as graph purturbation. 

- B.) Before generating the actual sequences we will follow the recently published (?) splat [(Zappia et al.)](#zappia) model. The generative process followed in splat involved learning model parameters from the experimental scRNA-seq data and then by using those model parameters genes/transcripts are repeatedly sampled. The raw _counts_ estimated in this fashion incorporated the main features of scRNA-seq protocols such as presence of outlier genes, gene dispersion, gene drop-out etc. These count act as proportion while selecting a gene/isoform to sequence from. Given an candidate gene/transcript, we will use the user defined protocol to (full length/ 3'-based etc) to choose the position to sample the sequence from. We will further implement seq-specific bias and GC-bias to emulate the same present in real life. 

- C.) (I want to add this but not sure:  (This will be about DE-genes how we can model normalization step and can generate data with polyester kind of feature, lot to say here, but have to be concise.)

#### Deliverables for Aim 2
The deliverable for aim 2 is a fully functional fragment level simulator that can take the real scRNA-seq dataset as input and learn various model parameters. Learned model parameters will be used to generate final gene level counts [(Zappia et al.)](#zappia). Depending on the protocol the set of UMIs will be generated. We will build the UMI-to-molecule graph and will incorporate user defined noise to create the final set of mapping. In the final step the simulator will generate the raw sequences after applying different bias criteria specified by user. Following majority of the bulk RNA-seq simulators, the name of the reads will contain the true origin (UMI and transcript). We intend to test various scRNA-seq quantification tools to benchmark the accuracy levels. The scRNA-seq atlas will be used as an input to the simulator to design a gold standard simulated dataset for future benchmarks. 



## References cited
---------
<a name="zappia">Zappia</a>, Luke, Belinda Phipson, and Alicia Oshlack. "Splatter: Simulation Of Single-Cell RNA Sequencing Data." bioRxiv (2017): 133173.

<a name="frazee">Frazee</a>, Alyssa C., et al. "Polyester: simulating RNA-seq datasets with differential transcript expression." Bioinformatics 31.17 (2015): 2778-2784.

<a name="smith">Smith</a>, Tom, Andreas Heger, and Ian Sudbery. "UMI-tools: modeling sequencing errors in Unique Molecular Identifiers to improve quantification accuracy." Genome research 27.3 (2017): 491-499.

<a name="tung">Tung</a>, Po-Yuan, et al. "Batch effects and the effective design of single-cell gene expression studies." Scientific reports 7 (2017): 39921.
